crates/cli/src/main.rs

```diff
@@ -378,37 +383,56 @@ async fn start_program() -> Result<(), AnvilZksyncError> {
     if config.use_evm_emulator {
         // We need to enable EVM emulator by setting `allowedBytecodeTypesToDeploy` in `ContractDeployer`
         // to `1` (i.e. `AllowedBytecodeTypes::EraVmAndEVM`).
-        let service_call_pseudo_caller =
-            Address::from_str("0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF").unwrap();
-        node.impersonate_account(service_call_pseudo_caller)
-            .unwrap();
-        node.set_rich_account(service_call_pseudo_caller, U256::from(1_000_000_000_000u64))
+        node.impersonate_account(PSEUDO_CALLER).unwrap();
+        node.set_rich_account(PSEUDO_CALLER, U256::from(1_000_000_000_000u64))
             .await;
-        let tx = L2TxBuilder::new(
-            service_call_pseudo_caller,
-            Nonce(0),
-            U256::from(300_000),
-            U256::from(u32::MAX),
-            node.chain_id().await,
-        )
-        .with_to(CONTRACT_DEPLOYER_ADDRESS)
-        .with_calldata(
-            // cast calldata "setAllowedBytecodeTypesToDeploy(uint8)()" 1
-            hex::decode("fe06380c0000000000000000000000000000000000000000000000000000000000000001")
-                .unwrap(),
-        )
-        .build_impersonated();
+        let chain_id = node.chain_id().await;
+        let mut txs = Vec::with_capacity(PREDEPLOYS.len() + 1);
+        txs.push(
+            L2TxBuilder::new(
+                PSEUDO_CALLER,
+                Nonce(0),
+                U256::from(300_000),
+                U256::from(u32::MAX),
+                chain_id,
+            )
+            .with_to(CONTRACT_DEPLOYER_ADDRESS)
+            .with_calldata(Bytes::from_static(ENABLER_CALLDATA).to_vec())
+            .build_impersonated()
+            .into(),
+        );
+
+        // If evm emulator is enabled, and not in fork mode, deploy pre-deploys for dev convenience
+        if !is_fork_mode {
```

Comment:
Hmm, this made me think - does it even make sense to allow `--emulate-evm` with `fork`? If base chain does not allow EVM bytecode there seems to be little reason to enable it. Also I wrote the code above with base case in mind so it might not be enough to enable EVM emulator for all possible fork configurations.

Reply:
Users should be able to fork (v27 supporting chains) and deploy EraVM and EVM contracts without passing any flag imo.

That would remove the `--emulate-evm` flag all together really? Not sure how much refactoring that would require though.

Reply:
> Users should be able to fork (v27 supporting chains) and deploy EraVM and EVM contracts without passing any flag imo.

Yep, that's what I am alluding to as well

> That would remove the --emulate-evm flag all together really?

Still useful for non forking mode I guess. We can consider enabling it by default once V27 becomes the default version (should happen soon).

> Not sure how much refactoring that would require though.

I think we just have to make a call to fork's `ContractDeployer` and see what bytecodes it supports and then change base system contract's options accordingly.